astroquery/esasky/core.py

```diff
@@ -75,6 +72,116 @@ class ESASkyClass(BaseQuery):
     _SPECTRA_DOWNLOAD_DIR = "Spectra"
     _isTest = ""
 
+    _NUMBER_DATA_TYPES = ["REAL", "float", "INTEGER", "int", "BIGINT", "long", "DOUBLE", "double", "SMALLINT", "short"]
+
+    def __init__(self, tap_handler=None):
+        super(ESASkyClass, self).__init__()
+
+        if tap_handler is None:
+            self._tap = TapPlus(url=self.URLbase + "/tap")
+        else:
+            self._tap = tap_handler
+
+    def query(self, query, *, output_file=None, output_format="votable", verbose=False):
+        """Launches a synchronous job to query the ESASky TAP
+
+        Parameters
+        ----------
+        query : str, mandatory
+            query (adql) to be executed
+        output_file : str, optional, default None
+            file name where the results are saved if dumpToFile is True.
+            If this parameter is not provided, the jobid is used instead
+        output_format : str, optional, default 'votable'
+            possible values 'votable' or 'csv'
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A table object
+        """
+        job = self._tap.launch_job(query=query, output_file=output_file,
+                                   output_format=output_format,
+                                   verbose=verbose,
+                                   dump_to_file=output_file is not None)
+        return job.get_results()
+
+    def get_tables(self, *, only_names=True, verbose=False):
```

Comment:
The first commit of adding TAP and ADQL functionality is indeed doesn't seem to be covered by tests. Could you please follow it up either here or in a new PR, if this is merged in the meantime?